lib/renderer/web-view/web-view-impl.js

```diff
@@ -246,16 +240,10 @@ const setupMethods = (WebViewElement) => {
 
   const createNonBlockHandler = function (method) {
     return function (...args) {
-      const callback = (typeof args[args.length - 1] === 'function') ? args.pop() : null
-      const requestId = getNextId()
-      ipcRendererInternal.once(`ELECTRON_GUEST_VIEW_MANAGER_ASYNC_CALL_RESPONSE_${requestId}`, function (event, error, result) {
-        if (error == null) {
-          if (callback) callback(result)
-        } else {
-          throw errorUtils.deserialize(error)
-        }
-      })
-      ipcRendererInternal.send('ELECTRON_GUEST_VIEW_MANAGER_ASYNC_CALL', requestId, getGuestInstanceId(this), method, args, callback != null)
+      ipcRendererUtils.invoke('ELECTRON_GUEST_VIEW_MANAGER_CALL', getGuestInstanceId(this), method, args)
+        .catch(error => {
+          throw error
```

Comment:
What's the intended behaviour on error here? because I think catch/throw is the same as not catching at all. the exception will be thrown in a subsequent microtask.

Reply:
hmm, does not make sense. removed.